### DISCUSSION TITLE
How to get the frequency response results of the acceleration of a NamedSelection node

### DISCUSSION THREAD

**Discussion by hrntsm** [ansys/pydpf-core#1279] [Q&A] [answered]
I am using Mechanical for frequency response analysis. I am trying to get the results using dpf.
The code is as follows.

```
sim = post.load_simulation("my_result.rst")
print(sim)
```

The sim's DPF model returned the following
Does it not include acceleration results?
When I checked with WB, I get results that are included.

```
DPF Model
------------------------------
Harmonic analysis
Unit system: NMM: mm, ton, N, s, mA, degC
Physics Type: Mechanical
Available results:
     -  displacement: Nodal Displacement
     -  reaction_force: Nodal Force
     -  stress: ElementalNodal Stress
     -  elemental_volume: Elemental Volume
     -  stiffness_matrix_energy: Elemental Energy-stiffness matrix
     -  artificial_hourglass_energy: Elemental Hourglass Energy
     -  thermal_dissipation_energy: Elemental thermal dissipation energy
     -  kinetic_energy: Elemental Kinetic Energy
     -  co_energy: Elemental co-energy
     -  incremental_energy: Elemental incremental energy
     -  elastic_strain: ElementalNodal Strain
     -  structural_temperature: ElementalNodal Temperature
------------------------------
```

Acceleration is preferred, but if it is not included, the frequency response of the displacement of the nodes selected in Named Selection is also acceptable.
How should I get it after this?

**Comment by PProfizi** [comment]
Hi @hrntsm,

It seems that DPF did not find acceleration results in the `RST` file. The reason you see it available in Mechanical may be that it recomputes it. It is possible it is not stored in the `RST` file. @ayush-kumar-423 do you know if acceleration results are stored in `RST` for frequency response analyses?

Now to get the displacement fields for the nodes in a Named Selection, you should be able to follow [this example](https://post.docs.pyansys.com/version/dev/examples/01-Detailed-Examples/01-named-selections-modal-simulation.html) despite it showcasing a modal analysis.

**Comment by ayush-kumar-423** [comment]
Hi,

@hrntsm acceleration results are not stored by default in the RST for a Harmonic Response Analysis. Mechanical calculates them as the second order derivative of displacement. For a Harmonic response, you can multiply the Displacement field with (`-omega^2`) to obtain acceleration. Where `omega` is Angular Velocity = `2*pi*f`.
For multiplication you can use `dpf.operators.math.scale()`

**Comment by hrntsm** [comment]
Thanks @PProfizi @ayush-kumar-423 !
 
I was able to retrieve the results.
I see that I can output the result on the terminal when I run the following code, but how can I make a graph of the frequency response or get the value of each frequency?
Is it possible to convert the results to a numpy array or something similar?
```
sim = post.HarmonicMechanicalSimulation(r"my_result.rst")
disp = sim.displacement(all_sets=True, named_selections=["C_100001"])
print(disp)
```

```
             results           U                                                             ...
             set_ids           1                       2                       3             ...
             complex           0           1           0           1           0           1 ...
 node_ids components                                                                         ...
   100001          X -1.1033e-04  2.9118e-06  1.6701e-08 -6.3268e-07 -4.7175e-05  3.2532e-03 ...
                   Y  3.7236e-04 -6.4921e-06 -8.8455e-05  1.9681e-06  3.7314e-06  2.1758e-03 ...
                   Z  2.5355e-03 -6.5708e-06  1.4849e-05 -7.9472e-06 -1.3825e-04 -1.4386e-02 ...
```

**Reply by PProfizi** [reply, marked as answer]
Hi @hrntsm,
You can indeed convert a `post.Dataframe` to a `np.array` as shown in [this example](https://post.docs.pyansys.com/version/dev/examples/01-Detailed-Examples/05-export_data.html). The caveat is that since NumPy arrays do not have column or row labels, you must first select a column of interest before exporting it.
As for plotting graphs, we do not have yet this capability native for the `post.Dataframe` so for now the best way is to perform selections and filtering on the `Dataframe`, then query `Dataframe.array`, and then plot this data using `Matplotlib` or `Plotly`.

**Reply by hrntsm** [reply]
Hi @PProfizi 

The results were retrieved and the process I was thinking about was done.
Thank you!
